docs/website/docs/dlt-ecosystem/verified-sources/rest_api.md

```diff
@@ -560,49 +560,81 @@ This will include the `id`, `title`, and `created_at` fields from the `issues` r
 
-    For example, in the `issues` resource configuration in the GitHub example, we have:
+### Incremental loading in `params`
 
-    ```py
-    {
-        "since": {
+Imagine we have the following endpoint `https://api.example.com/posts` and it:
+1. Accepts a `created_since` query parameter to fetch posts created after a certain date.
+2. Returns a list of posts with the `created_at` field for each post.
+
+For example, if we query the endpoint with `https://api.example.com/posts?created_since=2024-01-25`, we get the following response:
+
+```json
+{
+    "results": [
+        {"id": 1, "title": "Post 1", "created_at": "2024-01-26"},
+        {"id": 2, "title": "Post 2", "created_at": "2024-01-27"},
+        {"id": 3, "title": "Post 3", "created_at": "2024-01-28"}
+    ]
+}
+```
+
+To enable the incremental loading for this endpoint, you can use the following configuration:
+
+```py
+{
+    "path": "posts",
+    "data_selector": "results",  # Optional JSONPath to select the list of posts
+    "params": {
+        "created_since": {
             "type": "incremental",
-            "cursor_path": "updated_at",
-            "initial_value": "2024-01-25T11:21:28Z",
+            "cursor_path": "created_at", # The JSONPath to the field we want to track in each post
+            "initial_value": "2024-01-25",
         },
-    }
-    ```
+    },
+}
+```
 
-    This configuration tells the source to create an incremental object that will keep track of the `updated_at` field in the response and use it as a value for the `since` parameter in subsequent requests.
+After you run the pipeline, dlt will keep track of the last `created_at` from all the posts fetched and use it as the `created_since` parameter in the next request.
+So in our case, the next request will be made to `https://api.example.com/posts?created_since=2024-01-28` to fetch only the new posts created after `2024-01-28`.
 
-2. Specifying the `incremental` field in the [endpoint configuration](#endpoint-configuration):
+Now, let's break down the configuration. The `created_since` parameter is defined as an incremental parameter with the following fields:
 
-    ```py
-    {
-        "incremental": {
-            "start_param": "<parameter_name>",
-            "end_param": "<parameter_name>",
-            "cursor_path": "<path_to_cursor_field>",
-            "initial_value": "<initial_value>",
-            "end_value": "<end_value>",
-        }
-    }
-    ```
+```py
+{
+    "<parameter_name>": {
+        "type": "incremental",
+        "cursor_path": "<path_to_cursor_field>",
+        "initial_value": "<initial_value>",
+    },
+}
+```
+
+- `type`: The type of the incremental parameter. Set to `incremental`.
+- `cursor_path`: The JSONPath to the field within each item in the list that will be used as the cursor value. In this case, it's `created_at`. Note that the path starts from the root of the item (dict) and not from the root of the response.
```

Comment:
```suggestion
- `cursor_path`: The JSONPath to the field within each item in the list that will be used as the cursor value. In this case, it's `created_at`. Note that the path starts from the root of the item (dict) and not from the root of the response.

`{
    "responses": [
        {
            "submissionId": "ggg",
            "submissionTime": "2024-04-17T20:09:38.716Z",
            "lastUpdatedAt": "2024-04-17T20:09:38.716Z",
        },
        {
            "submissionId": "yyy",
            "submissionTime": "2024-04-17T20:11:38.716Z",
            "lastUpdatedAt": "2024-04-17T20:11:38.716Z",
        }
    ]
}`
For example, in the response above, and the cursor path is inside the array.
Thus, we should modify the `data_selector` parameter to "responses", and the `cursor_path` to "submissionTime"
`                "endpoint": {
                    "data_selector": "responses",
                    "params": {
                        "since": {
                            "type": "incremental",
                            "cursor_path": "submissionTime",
                            "initial_value": "2022-01-01T11:21:28Z",
                        },
                    },
                },`
```

Reply:
Hey @omri-antman-imagen, thanks for the suggestion. I believe introducing a second JSON example would be confusing for the reader, so I suggest using only one example (e.g. "posts"). Nevertheless, I think I get your point, so I adjusted the text to really stress that the `cursor_path` is within the data item and added an example to illustrate that in commit 70e43afc94719ec428730961827205e3e547ed31. Please take a look.